config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Policy.java

```diff
@@ -168,4 +174,27 @@ public Policy build() {
             return new Policy(this);
         }
     }
+
+    static void checkRepeatedTags(Bundle bundle, PolicyType policyType) {
+        Set<String> errors = new HashSet<>();
+        bundle.getPolicies().values()
+                .stream()
+                .filter(p -> p.getTag() != null)
+                .collect(groupingBy(Policy::getTag, Collectors.mapping(identity(), toList())))
+                .forEach((key, value) -> {
+                    if (value.size() > 1) {
+                        errors.add(
+                                String.format(
+                                        "Found more then one %s policy with tag '%s': [%s]",
+                                        policyType.getType(),
+                                        key,
+                                        Joiner.on(", ").join(value.stream().map(Policy::getPath).collect(toList()))
+                                )
+                        );
+                    }
+                });
+        if (!errors.isEmpty()) {
+            throw new ConfigLoadException(Joiner.on("\n").join(errors));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use "String.join" or "java.util.stream.Collectors.joining" instead. ([squid:S4738](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS4738))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=128&issues=AWcPsqq3d6oxh3R491C7&open=AWcPsqq3d6oxh3R491C7)

[ISSUE_KEY]: AWcPsqq3d6oxh3R491C7